common-api/src/main/java/org/pmiops/workbench/cdr/model/Concept.java

```diff
@@ -167,6 +184,21 @@ public Concept prevalence(float prevalence) {
         return this;
     }
 
+    @OneToMany(fetch = FetchType.LAZY, orphanRemoval = true, cascade = CascadeType.ALL, mappedBy = "concept")
```

Comment:
Since this collection is fetched lazily, I believe by default we're going to wind up doing a separate database fetch for the list of synonyms for every concept returned in a response.... you could confirm this by turning on SQL debugging. We need to make sure these get joined in / fetched eagerly when returning concept search results to avoid this.

Reply:
Also: what's the maximum number of synonyms for a given concept? 

We should make sure we don't try to return, say, hundreds or thousands of synonyms for a concept in a response, as that's not useful.

Reply:
The one with most is  Molecular pathology procedure, Level 2. It has 204.  
Next is 94. Level 5. Then it gets down to 20. So 204 concepts have it over 20.  and 79 concepts have more than 30.   

The plan is to hide them in the Ui in an expanding text area or scrolling list. Or maybe we just show the synonyms that match the search in the databrowser. At any rate, i think that the api 
should return all of them and the ui's can do what they want.  Or maybe just return synonyms that match the search ? 

With a bunch of synonyms many are almost exactly a like. For example there are 49 synonyms for male erectile disorder i noticed in my search.  "Sexual impotence" and "Sexual impotence, Male" and  "Male Sexual impotence "  . But for the procedures like 42742371 that has 204 synonms, the synonyms look much more significantly different.   

So this is a problem we will have to tackle somehow. I think for now though just hiding or limiting the list in the ui is the best way to go since all this may change in the future.

Reply:
I just checked and the concept synonyms seem to be coming back.

Reply:
Changed the fetch type to lazy and used join fetch. Things seem to be working fine. JPA does not now query concept synonym table each time.

Reply:
Fantastic!